crates/core/src/node/network_bridge/p2p_protoc.rs

```diff
@@ -682,24 +702,51 @@ impl P2pConnManager {
             .pending_from_executor
             .remove(msg.id())
             .then(|| executor_listener.callback());
-        let pending_client_req = state
-            .tx_to_client
-            .get(msg.id())
-            .cloned()
-            .map(|clients| clients.into_iter().collect::<Vec<_>>())
-            .or(state
-                .client_waiting_transaction
-                .iter_mut()
-                .find_map(|(tx, clients)| match (&msg, &tx) {
-                    (
-                        NetMessage::V1(NetMessageV1::Subscribe(SubscribeMsg::ReturnSub {
-                            key,
-                            ..
-                        })),
-                        WaitingTransaction::Subscription { contract_key },
-                    ) if contract_key == key.id() => Some(clients.drain().collect::<Vec<_>>()),
-                    _ => None,
-                }));
+        // For UPDATE operations, add extra logging and ensure proper state management
+        let is_update_op = msg.id().transaction_type().to_string().contains("Update");
+
+        let pending_client_req = if is_update_op {
+            // For UPDATE operations, remove clients from tracking to prevent stale state
+            let clients = state
```

Comment:
I very much doubt this is not breaking something, what happens after the update completes? we lost the subscription on the ws side cause those connections are destroyed? at the very least they should be returned to the pool? need to think this throught